activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -690,6 +690,28 @@ def rewhere(conditions)
       scope
     end
 
+    # Allows you to invert an entire where clause instead of manually applying conditions.
+    #
+    #   class User
+    #     scope :active, -> { where(accepted: true, locked: false) }
+    #   end
+    #
+    #   User.where(accepted: true)
+    #   # WHERE `accepted` = 1
+    #
+    #   User.where(accepted: true).invert_where
+    #   # WHERE `accepted` != 1
+    #
+    #   User.active
+    #   # WHERE `accepted` = 1 AND `locked` = 0
+    #
+    #   User.active.invert_where
+    #   # WHERE NOT (`accepted` = 1 AND `locked` = 0)
+    def invert_where
+      self.where_clause = where_clause.invert
+      self
```

Comment:
```suggestion
      scope = spawn
      scope.where_clause = where_clause.invert
      scope
```

We don't change relation inline for public API methods in Relation.

Reply:
Sure! I'll fix that.

While I'm here, would you prefer `invert_where` or `where.invert`? I saw `where.missing` and `where.associated` made it in, so maybe that would fit better.

Reply:
Since it will not take any argument I'd prefer `invert_where`

Reply:
Updated! I used the existing examples to do an `invert_where` that does `spawn.invert_where!` and an `invert_where!` that modifies in place.

Reply:
> Since it will not take any argument I'd prefer `invert_where`

That would also match `rewhere`. It's closer to manipulating the `where` clause itself than doing extra operations or expressions on it, like `where.not` etc.